src/rules/no_node_globals.rs

```diff
@@ -0,0 +1,308 @@
+// Copyright 2020-2024 the Deno authors. All rights reserved. MIT license.
+use super::program_ref;
+use super::Context;
+use super::LintRule;
+use crate::diagnostic::LintFix;
+use crate::diagnostic::LintFixChange;
+use crate::handler::Handler;
+use crate::handler::Traverse;
+use crate::Program;
+use std::borrow::Cow;
+
+use deno_ast::view as ast_view;
+use deno_ast::SourcePos;
+use deno_ast::SourceRange;
+use deno_ast::SourceRanged;
+use deno_ast::SourceRangedForSpanned;
+
+#[derive(Debug)]
+pub struct NoNodeGlobals;
+
+const CODE: &str = "no-node-globals";
+const MESSAGE: &str = "NodeJS globals are not available in Deno";
+const IMPORT_HINT: &str = "Import from the appropriate module instead";
+const IMPORT_FIX_DESC: &str = "Replace node global with module import";
+const REPLACE_FIX_DESC: &str = "Replace node global with corresponding value";
+const REPLACE_HINT: &str = "Use the corresponding value instead";
+
+static NODE_GLOBALS: phf::Map<&'static str, FixKind> = phf::phf_map! {
+  "process" => FixKind::Import { module: "node:process", import: "process" },
+  "Buffer" => FixKind::Import { module: "node:buffer", import: "{ Buffer }" },
+  "global" => FixKind::Replace("globalThis"),
+  "setImmediate" => FixKind::Import { module: "node:timers", import: "{ setImmediate }" },
+  "clearImmediate" => FixKind::Import { module: "node:timers", import: "{ clearImmediate }" },
+};
+
+impl LintRule for NoNodeGlobals {
+  fn lint_program_with_ast_view<'view>(
+    &self,
+    context: &mut Context<'view>,
+    program: Program<'view>,
+  ) {
+    NoNodeGlobalsHandler {
+      most_recent_import_range: None,
+    }
+    .traverse(program, context);
+  }
+
+  fn code(&self) -> &'static str {
+    CODE
+  }
+
+  fn tags(&self) -> &'static [&'static str] {
+    &["recommended"]
+  }
+
+  #[cfg(feature = "docs")]
+  fn docs(&self) -> &'static str {
+    include_str!("../../docs/rules/no_node_globals.md")
+  }
+}
+
+struct NoNodeGlobalsHandler {
+  most_recent_import_range: Option<SourceRange>,
+}
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+enum FixKind {
+  Import {
+    module: &'static str,
+    import: &'static str,
+  },
+  Replace(&'static str),
+}
+
+#[derive(Default)]
+enum AddNewline {
+  Leading,
+  Trailing,
+  #[default]
+  None,
+}
+
+impl FixKind {
+  fn hint(&self) -> &'static str {
+    match self {
+      FixKind::Import { .. } => IMPORT_HINT,
+      FixKind::Replace(_) => REPLACE_HINT,
+    }
+  }
```

Comment:
Nitpick: Can you go through and add blank lines between the methods? I wish rustfmt would do it.